### PR TITLE
Bluetooth: audio: ascs: Handle CIS failed to be established error

### PR DESCRIPTION
**Pull request by MariuszSkamra**
In case CIS failed to be established, leave the ASE in enabling state waiting for peer to retry.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

### CODE REVIEW

**Review by Thalley — DISMISSED** [review]

**Comment by MariuszSkamra** [comment]
Rebase

**Comment by Thalley** [comment]
> Rebase

Btw, you can restart tests (which does a rebase for test purposes) without rebasing the PR (cauisng reviews to be dismissed)

**Review by Thalley — APPROVED** [review]

File: subsys/bluetooth/audio/ascs.c [hunk 1]
```diff
@@ -541,6 +542,18 @@ static void ascs_ep_iso_disconnected(struct bt_audio_ep *ep, uint8_t reason)
 
 	LOG_DBG("stream %p ep %p reason 0x%02x", stream, stream->ep, reason);
 
+	if (ep->status.state == BT_AUDIO_EP_STATE_ENABLING &&
+	    reason == BT_HCI_ERR_CONN_FAIL_TO_ESTAB) {
+		LOG_DBG("Waiting for retry");
+
+		err = bt_audio_stream_iso_listen(stream);
```
[the remark this answers is not in the document]
Comment (Thalley) [reply subsys/bluetooth/audio/ascs.c:549]: https://github.com/zephyrproject-rtos/zephyr/issues/51552 may be related

[the remark this answers is not in the document]
Comment (fredrikdanebjer) [reply subsys/bluetooth/audio/ascs.c:549]: Cheers, I got it now!

**Review by fredrikdanebjer — APPROVED** [review]

**Pull request merged**